Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -915,6 +915,10 @@ public void SetHoldings(Symbol symbol, decimal percentage, bool liquidateExistin
             {
                 MarketOrder(symbol, quantity, false, tag);
             }
+            else
+            {
+                Error($"Unable to SetHoldings('{symbol.Value}', {percentage}) because CalculateOrderQuantity returned zero quantity.");
```

Comment:
Sometimes decimal values end up with arbitrarily long precision, recommend using `percentage.Normalize()` to remove trailing zeros from the precision.